File: frontend/src/entities/choir.ts
import { Entity, EntityRef } from "./entity"
import { LiturgicalBook } from "./liturgicalBook"
import { Rite } from "./rite"
import { User } from "./user"

export interface Choir extends Entity {
    id: number
    name: string
    type: string
    inviteToken?: string
    manager?: EntityRef<User>
    rite?: EntityRef<Rite>
    liturgicalBook?: EntityRef<LiturgicalBook>
}

File: backend/src/main/kotlin/nl/stevenbontenbal/chorister/domain/users/Choir.kt
package nl.stevenbontenbal.chorister.domain.users

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import nl.stevenbontenbal.chorister.domain.rites.Rite

@Entity
class Choir(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var name: String,
    var type: String,
    var inviteToken: String? = null,
    @OneToMany(mappedBy = "choir")
    @JsonIgnoreProperties("choir")
    var users: MutableList<User>? = mutableListOf(),
    @OneToOne(cascade = [CascadeType.ALL])
    @JsonIgnoreProperties("choir")
    var manager: User?,
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "RITE_ID")
    var rite: Rite?
)
{
    companion object
}